### PR TITLE
Handle more fields in plugin meta YAML

### PR DESCRIPTION
**Pull request by mkuznyetsov**
### What does this PR do?
Add more possible fields for plugin yaml (not validated anyhow atm), for use in plugin viewer:
category
publisher
repository
tags
mediaImage
mediaVideo
firstPublicationDate
latestUpdateDate
preview

### What issues does this PR fix or reference?
https://github.com/eclipse/che-plugin-registry/issues/71

### CODE REVIEW

**Review by skabashnyuk — COMMENTED** [review]
LGTM @garagatyi wdyt?

File: wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/model/PluginMeta.java [hunk 1]
```diff
@@ -109,4 +118,85 @@ public PluginMeta attributes(Map<String, String> attributes) {
...
+  public String getMediaImage() {
+    return mediaImage;
+  }
+
+  public PluginMeta mediaVideo(String mediaVideo) {
+    this.mediaVideo = mediaVideo;
+    return this;
+  }
+
+  public String getediaVideo() {
```
Comment (sleshchenko) [inline wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/model/PluginMeta.java]: I guess there is typo in this method name

Comment (mkuznyetsov) [reply wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/model/PluginMeta.java]: ty

File: wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/model/PluginMeta.java [hunk 2]
```diff
@@ -109,4 +118,85 @@ public PluginMeta attributes(Map<String, String> attributes) {
...
+  public PluginMeta repository(String repository) {
+    this.repository = repository;
+    return this;
+  }
+
+  public String getRepository() {
+    return repository;
+  }
+
+  public PluginMeta tags(String tags) {
```
Comment (sleshchenko) [inline wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/model/PluginMeta.java]: Should not it be a list of strings?

Comment (mkuznyetsov) [reply wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/model/PluginMeta.java]: yes, I guess it should :+1:

File: wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/model/PluginMeta.java [hunk 3]
```diff
@@ -109,4 +119,85 @@ public PluginMeta attributes(Map<String, String> attributes) {
...
+    return repository;
+  }
+
+  public PluginMeta tags(List<String> tags) {
+    this.tags = tags;
+    return this;
+  }
+
+  public List<String> getTags() {
+    return tags;
```
Comment (garagatyi) [inline wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/model/PluginMeta.java:162]: Possible NPE. We do not return empty collections.

File: wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/model/PluginMeta.java [hunk 4]
```diff
@@ -22,6 +23,15 @@
   private String version = null;
   private String title = null;
   private String description = null;
+  private String category = null;
+  private String publisher = null;
+  private String repository = null;
+  private List<String> tags = null;
+  private String mediaImage = null;
```
Comment (garagatyi) [inline wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/model/PluginMeta.java:33]: Serialization is incorrect. Should be `media-img`

File: wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/model/PluginMeta.java [hunk 5]
```diff
@@ -22,6 +23,15 @@
...
+  private String mediaVideo = null;
```
Comment (garagatyi) [inline wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/model/PluginMeta.java:34]: Serialization is incorrect. Should be `media-video`

File: wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/model/PluginMeta.java [hunk 6]
```diff
@@ -22,6 +23,15 @@
...
+  private String firstPublicationDate = null;
```
Comment (garagatyi) [inline wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/model/PluginMeta.java:35]: @slemeur is it OK that we have an inconsistent style of fields names in the YAML? 
`media-video` with dash
`firstPublicationDate` with camel case

**Review by garagatyi — CHANGES_REQUESTED** [review]

**Comment by mkuznyetsov** [comment]
ci-test

File: wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/model/PluginMeta.java (hunk 3 above)
Comment (mkuznyetsov) [reply wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/model/PluginMeta.java:162]: fixed

File: wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/model/PluginMeta.java [hunk 7]
```diff
@@ -153,6 +154,9 @@ public PluginMeta tags(List<String> tags) {
   }
 
   public List<String> getTags() {
+    if (tags == null) {
+      return new ArrayList<>();
```
Comment (garagatyi) [inline wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/model/PluginMeta.java]: It should be assigned

Comment (mkuznyetsov) [reply wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsplugins/model/PluginMeta.java]: fixed, also added initialization in the field itself

**Review by garagatyi — APPROVED** [review]

**Review by sleshchenko — APPROVED** [review]

**Comment by mkuznyetsov** [comment]
@garagatyi fyi, as @sleshchenko said, I decided to remove initialization in all fields in the class, since they are lazily initialized in getters anyway, and so would be consistent for all fields

**Review by skabashnyuk — APPROVED** [review]

**Comment by skabashnyuk** [comment]
@ibuziuk FYI

**Comment by mkuznyetsov** [comment]
ci-test

**Comment by mkuznyetsov** [comment]
@garagatyi @sleshchenko I have also a proposition to add `@JsonIgnoreProperties(ignoreUnknown=true)`
annotation, so that we would be able to allow to have fields in plugin yamls, that are not defined in model class. This may prevent Che from failing to start workspaces, which ends up using plugin yaml that may have such fields, in case there will be new ones added in future

**Comment by garagatyi** [comment]
Please, sign your commits.
I'm OK with the suggested solution and OK to do it in this PR or another one.

**Comment by sleshchenko** [comment]
> I have also a proposition to add @JsonIgnoreProperties(ignoreUnknown=true)

I'm OK with your proposal.

**Comment by mkuznyetsov** [comment]
ci-test

**Comment by artaleks9** [comment]
Selenium tests execution on Eclipse Che Multiuser on OCP (https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/1506/) doesn't show any regression against this Pull Request.

**Pull request merged**